scripts/qemu.py

```diff
@@ -110,8 +114,19 @@ def _base_args(self):
                 self._monitor_address[1])
         else:
             moncdev = 'socket,id=mon,path=%s' % self._monitor_address
+
+        if isinstance(self._console_address, tuple):
+            consoledev = "socket,id=console,host=%s,port=%s" % (
+                self._console_address[0],
+                self._console_address[1])
+        else:
+            consoledev = ('socket,id=console,path=%s,server,nowait' %
+                          self._console_address)
+
         return ['-chardev', moncdev,
                 '-mon', 'chardev=mon,mode=control',
+                '-chardev', consoledev,
+                '-device', 'isa-serial,chardev=console',
```

Comment:
Also I'd not support list of arguments, but rather I'd expect free-form extra_arguments).